#164139648 Implement functionality to create articles

codinger41 opened the pull request:
#### What does this PR do?
Adds the feature for users to be able to create articles
#### Description of Task to be completed?
Users should be able to create articles and before creating articles, they should be able to preview the articles
In this task, I did the following:
- styled article creation page with preview option
- added action to make a request to create articles

#### How should this be manually tested?
- Start the server and visit `localhost:3000/create-article`
- N/A
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/164139648

#### Screenshots (if appropriate)
#### Questions:
N/A

codinger41 committed:
feat(create-articles): consume API endpoint for article creation
- style article creation page with preview option
- add action to make request to create articles on the backend

[Finishes #164139648]

codinger41 committed:
feat(create-articles): consume API endpoint for article creation
- style article creation page with preview option
- add action to make request to create articles on the backend

[Finishes #164139648]

9jaswag commented:
@leksyib you still haven't changed the PR description

9jaswag commented on package.json:
```diff
@@ -38,7 +40,9 @@
     "axios": "^0.18.0",
     "bootstrap": "^4.3.1",
     "express": "^4.16.4",
+    "form-data": "^2.3.3",
```
@leksyib why do we need this?

codinger41 replied:
@9jaswag  The actual FormData API was giving me some issues, also I had to get `boundary` while generating a request header but the FormData API wasn't giving the boundary for the header.

9jaswag replied:
Also why are we using the `function` keyword?

codinger41 replied:
I was flagged seriously by hound for using `const` but I think it has been fixed. I'll make changes now.

9jaswag commented on src/__tests__/components/preview.spec.js:
```diff
@@ -0,0 +1,24 @@
+import { mount } from 'enzyme';
+import Preview from '../../components/previewArticle/Preview';
+
+describe('Preview shallow tests', () => {
```
@leksyib I passed a message about how we should name our describe blocks

9jaswag commented on src/components/previewArticle/Preview.jsx:
```diff
@@ -0,0 +1,50 @@
+          <ArticleTitle title={article.title || ''} />
+          <img
+              alt="image"
+              src={article.image ? URL.createObjectURL(article.image) : 'https://via.placeholder.com/700x400'}
```
@leksyib save this to a variable and use it here..to make things cleaner

9jaswag commented on src/views/createArticle/index.jsx:
```diff
@@ -0,0 +1,126 @@
+  }
+
+  function deleteArticle() {
+    window.location = '/create-article';
```
@leksyib why are we doing this?

codinger41 replied:
We want to clear the state and have a fresh page so I refreshed the page. What do you think will work best here?

9jaswag replied:
@leksyib A React application should be a SPA...meaning it should not reload. What does the function do?

9jaswag commented on src/views/createArticle/index.jsx:
```diff
@@ -0,0 +1,126 @@
+    return completeArticleCreation();
+  }
+
+  function updateInput(e) {
```
@leksyib please use `event` instead of `e`

codinger41 replied:
Thanks will do.

9jaswag reviewed (changes requested):
I also have not seen a screenshare for this

codinger41 committed:
feat(create-articles): consume API endpoint for article creation

- style article creation page with preview option
- add action to make request to create articles on the backend

[Finishes #164139648]

codinger41 committed:
feat(create-articles): consume API endpoint for article creation

- style article creation page with preview option
- add action to make request to create articles on the backend

[Finishes #164139648]

9jaswag commented on .hound.yml:
```diff
@@ -1,9 +1,5 @@
- sass-lint:
-  enabled: true
-  config_file: .sasslintrc.json 
+  
```
@leksyib there's an EOL prob here

codinger41 replied:
@9jaswag  fixed
